File: app/src/main/java/cn/bproject/neteasynews/adapter/PicListAdapter.java
package cn.bproject.neteasynews.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.aspsine.irecyclerview.IViewHolder;
import com.bumptech.glide.Glide;

import java.util.ArrayList;

import cn.bproject.neteasynews.R;
import cn.bproject.neteasynews.bean.PicListBean;

/**
 * Created by liaozhoubei on 2017/1/7.
 */

public class PicListAdapter extends RecyclerView.Adapter<PicListAdapter.ViewHolder> {

    private Context mContext;
    private ArrayList<PicListBean> mPicListBeens;
    private OnItemClickListener mOnItemClickListener;

    public PicListAdapter(Context context, ArrayList<PicListBean> picListBeens) {
        mContext = context;
        mPicListBeens = picListBeens;

    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = View.inflate(mContext, R.layout.item_pic_layout, null);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        PicListBean picListBean = mPicListBeens.get(position);

        String imageSrc = picListBean.getCover();
        String title = picListBean.getSetname();

        holder.tv_title.setText(title);

        Glide.with(mContext)
                .load(imageSrc)
                .placeholder(R.drawable.defaultbg)
                .crossFade()
                .into(holder.iv_pic);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int position = holder.getIAdapterPosition();
                final PicListBean picListBean = mPicListBeens.get(position);
                if (mOnItemClickListener != null) {
                    mOnItemClickListener.onItemClick(position, picListBean, v);
                }
            }
        });

    }

    @Override
    public int getItemCount() {
        return mPicListBeens.size();
    }

    /**
     * 设置Item点击监听
     *
     * @param listener
     */
    public void setOnItemClickListener(OnItemClickListener listener) {
        this.mOnItemClickListener = listener;
    }


    public interface OnItemClickListener<T> {
        void onItemClick(int position, T t, View v);
    }

    public class ViewHolder extends IViewHolder {

        public LinearLayout rl_root;
        public ImageView iv_pic;
        public TextView tv_title;


        public ViewHolder(View itemView) {
            super(itemView);
            rl_root = (LinearLayout) itemView.findViewById(R.id.rl_root);
            iv_pic = (ImageView) itemView.findViewById(R.id.iv_pic);
            tv_title = (TextView) itemView.findViewById(R.id.tv_title);
        }
    }


}

File: app/src/main/java/cn/bproject/neteasynews/adapter/PicListAdapter.kt
package cn.bproject.neteasynews.adapter

import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.paging.PagedListAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import cn.bproject.neteasynews.R
import cn.bproject.neteasynews.activity.PicDetailActivity

import cn.bproject.neteasynews.bean.PicListBean
import com.bumptech.glide.Glide

/**
 * 不明原因的会偶发的不显示界面，暂时不使用 PagedListAdapter
 */
class PicListAdapter(var tid:String) : PagedListAdapter<PicListBean, PicListAdapter.ViewHolder>(REPO_COMPARATOR) {
    private val BIG_IMG = 0
    private val SMALL_IMG = 1
    private val THREE_IMG = 2
    private val KEY_TID = "TID"  //频道id
    private val KEY_COLUMN = "COLUMN"
    private val SETID = "SETID"  // 图集id

    init {
        Log.e("PicListFragment", "PicListAdapter 获取信息${tid}  column  :")
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        var view: View = View.inflate(parent.context, R.layout.item_pic_layout, null)
        Log.e("PicListFragment", "PicListAdapter data ${getItem(0).toString()} :")
        return ViewHolder(view)
    }



    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val picListBean = getItem(position)

        val imageSrc = picListBean?.cover
        val title = picListBean?.setname

        holder.tv_title.setText(title)

        Glide.with(holder.iv_pic.context)
            .load(imageSrc)
            .placeholder(R.drawable.defaultbg)
            .into(holder.iv_pic)

        holder.itemView.setOnClickListener(){
            val id = picListBean!!.setid
            val intent = Intent(holder.iv_pic.context, PicDetailActivity::class.java)
            intent.putExtra(KEY_TID, tid)
            intent.putExtra(SETID, id)
            holder.iv_pic.context.startActivity(intent)
        }
    }

    private fun setNetPicture(url: String?, img: ImageView) {
        Glide.with(img.context)
            .load(url)
            .placeholder(R.drawable.defaultbg)
            .into(img)
    }

    companion object {
        private val REPO_COMPARATOR = object : DiffUtil.ItemCallback<PicListBean>() {
            override fun areItemsTheSame(oldItem: PicListBean, newItem: PicListBean): Boolean  {
                Log.e("PicListFragment", "PicListAdapter areItemsTheSame oldItem.setname${ oldItem.setname}  ")
                return oldItem.setname == newItem.setname
            }


            override fun areContentsTheSame(oldItem: PicListBean, newItem: PicListBean): Boolean {
                Log.e("PicListFragment", "PicListAdapter areContentsTheSame oldItem.setname${ oldItem.setname}  ")
                return   oldItem.setname == newItem.setname
            }

        }
    }

    class ViewHolder(var view: View) : RecyclerView.ViewHolder(view) {

        var rl_root: LinearLayout = view.findViewById(R.id.rl_root)
        var iv_pic: ImageView = view.findViewById(R.id.iv_pic);
        var tv_title: TextView = view.findViewById(R.id.tv_title);
    }
}